packages/core/src/components/hotkeys/hotkeysEvents.ts

```diff
@@ -53,14 +53,20 @@ export class HotkeysEvents {
     }
 
     public handleKeyDown = (e: KeyboardEvent) => {
-        if (this.isTextInput(e) || isHotkeysDialogShowing()) {
+        if (this.isTextInput(e)) {
             return;
         }
 
         const combo = getKeyCombo(e);
 
         if (comboMatches(parseKeyCombo(SHOW_DIALOG_KEY), combo)) {
-            showHotkeysDialog(this.actions.map((action) => action.props));
+            if (isHotkeysDialogShowing()) {
+                hideHotkeysDialogAfterDelay();
```

Comment:
Because a bunch of disjoint hotkey `keydown` listeners will fire in succession. If you synchronously call `hideHotkeysDialog` in any one of them, then immediate successive calls to `isHotkeysDialogShowing` would return `false`, causing the hotkeys dialog to reopen.

Introducing a small delay is the only way to guarantee all those `keydown` listeners execute before the dialog actually gets hidden.